File: examples/java/core/child-contexts/process-region-a.java
var data = (Map<String, Object>) event.get("data");

// Execute child contexts sequentially for each region
var resultA = ctx.runInChildContext("region_a", Map.class, childCtx -> {
    var result = childCtx.step("process_a", Map.class,
        stepCtx -> processForRegion("A", data));
    return Map.of("region", "A", "result", result);
});

var resultB = ctx.runInChildContext("region_b", Map.class, childCtx -> {
    var result = childCtx.step("process_b", Map.class,
        stepCtx -> processForRegion("B", data));
    return Map.of("region", "B", "result", result);
});

var resultC = ctx.runInChildContext("region_c", Map.class, childCtx -> {
    var result = childCtx.step("process_c", Map.class,
        stepCtx -> processForRegion("C", data));
    return Map.of("region", "C", "result", result);
});

return Map.of(
    "regions_processed", 3,
    "results", List.of(resultA, resultB, resultC));

File: examples/python/operations/child-contexts/process-region-a.py
@durable_with_child_context
def process_region_a(ctx: DurableContext, data: dict) -> dict:
    """Process data for region A."""
    result = ctx.step(lambda _: process_for_region("A", data), name="process_a")
    return {"region": "A", "result": result}

@durable_with_child_context
def process_region_b(ctx: DurableContext, data: dict) -> dict:
    """Process data for region B."""
    result = ctx.step(lambda _: process_for_region("B", data), name="process_b")
    return {"region": "B", "result": result}

@durable_with_child_context
def process_region_c(ctx: DurableContext, data: dict) -> dict:
    """Process data for region C."""
    result = ctx.step(lambda _: process_for_region("C", data), name="process_c")
    return {"region": "C", "result": result}

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Process data for multiple regions sequentially."""
    data = event["data"]
    
    # Execute child contexts sequentially
    result_a = context.run_in_child_context(
        process_region_a(data),
        name="region_a"
    )
    
    result_b = context.run_in_child_context(
        process_region_b(data),
        name="region_b"
    )
    
    result_c = context.run_in_child_context(
        process_region_c(data),
        name="region_c"
    )
    
    return {
        "regions_processed": 3,
        "results": [result_a, result_b, result_c],
    }
